### PR TITLE
review username tagging in messages

### PR DESCRIPTION
**Pull request by ritvikjain99**
Now we can tag user using just '@' !
@willemarcel @xamanu  please reveiw !
Related #2247

### CODE REVIEW

**Comment by ritvikjain99** [comment]
@willemarcel I noticed that on frontend it's just conversion of @[username] -> **username**, without the check whether the user was actually tagged. Which needs to be changed !
I am working on it, will make a PR soon :)

**Comment by ritvikjain99** [comment]
@willemarcel please review !

File: tests/backend/unit/services/messaging/test_messaging_service.py [hunk 1]
```diff
@@ -7,12 +7,14 @@ class TestMessagingService(unittest.TestCase):
     def test_message_service_identifies_all_users(self):
         # Act
         usernames = MessageService._parse_message_for_username(
-            'Hello @[Iain Hunter] and "[LindaA1]'
+            'Hello @Iain Hunter and "@LindaA1 @@@Cersie @ @Peter_123'
         )
 
         # Assert
-        self.assertEqual(usernames[0], "Iain Hunter")
+        self.assertEqual(usernames[0], "Iain")
```
Comment (xamanu) [inline tests/backend/unit/services/messaging/test_messaging_service.py]: Are we dropping support for usernames with a space in between? Like `Iain Hunter`?

**Comment by ritvikjain99** [comment]
@xamanu As far as I have seen usernames don't have spaces, and underscores are the substitutes for spaces, the reasons I see are:
1. We'll face problem sending username as query parameter.
For instance I have username "ritvik rajesh jain":
Then param would have to be: ?username=ritvik%20rajesh%20jain, which is not so readable.
2. For instance there are three persons having usernames differing just by whitespaces you can't differentiate between two people based on number of white spaces they have in their usernames.
E.g. "   ritvik", "  ritvik", "ritvik ", "ritvik    " , "hobs shaw", "hobs   shaw", "hobs  shaw" , all look the same, yet are different, they are really hard to track.
3. If someone uses tab('\t') in username then it would be really problematic as tab size can be altered.

Please correct me if my arguments are wrong !

**Comment by willemarcel** [comment]
I'm solving it in another PR

**Pull request closed without merging**